lib/Twig/Extension/Core.php

```diff
@@ -852,6 +852,39 @@ function twig_in_filter($value, $compare)
     return false;
 }
 
+/** 
+ * Returns a trimmed string.
+ * 
+ * @param string $string
+ * @param string $characterMask
+ * @param string $mode
+ *
+ * @return string
+ *
+ * @throws Twig_Error_Runtime When an type is not a string.
+ * @throws Twig_Error_Runtime When an invalid type is used (not 'left', 'right' or 'both')
+ */
+function twig_trim_filter($string, $characterMask = null, $type = 'both')
+{
+    if (!is_string($type)) {
+        throw new Twig_Error_Runtime(sprintf('Trimming type must be a string but got %s.', gettype($type)));
+    }
+    if (!in_array($type, ['both', 'left', 'right'])) {
```

Comment:
I would throw the exception in the `default` of the switch, as most of the time this will be false it is better for speed to not test it every time on forhand